workflow/controller/operator.go

```diff
@@ -2186,3 +2205,67 @@ func (woc *wfOperationCtx) runOnExitNode(parentName, templateRef, boundaryID str
 	}
 	return false, nil, nil
 }
+
+func (woc *wfOperationCtx) createPDBResource() error {
+
+	if woc.wf.Spec.PodDisruptionBudget == nil {
+		return nil
+	}
+
+	pdb, err := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Get(woc.wf.Name, metav1.GetOptions{})
+	if err != nil && !apierr.IsNotFound(err) {
+		return err
+	}
+	if pdb != nil && pdb.Name != "" {
+		return nil
+	}
+
+	pdbSpec := *woc.wf.Spec.PodDisruptionBudget
+	if pdbSpec.Selector == nil {
+		pdbSpec.Selector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{common.LabelKeyWorkflow: woc.wf.Name},
+		}
+	}
+
+	newPDB := policyv1beta.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   woc.wf.Name,
+			Labels: map[string]string{common.LabelKeyWorkflow: woc.wf.Name},
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(woc.wf, wfv1.SchemeGroupVersion.WithKind(workflow.WorkflowKind)),
+			},
+		},
+		Spec: pdbSpec,
+	}
+	_, err = woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Create(&newPDB)
+	if err != nil {
+		return err
+	}
+	woc.log.Infof("Created PDB resource for workflow.")
+	woc.updated = true
+	return nil
+}
+
+func (woc *wfOperationCtx) deletePDBResource() error {
```

Comment:
I don't think you should delete the PDB it here. It'll just get deleted when the workflow gets deleted due to the ownership reference.

Reply:
I think PDB doesn't hold any workflow related information to keep for until workflow delete. This is like a `DB transaction`. It prevents the pod deletion during workflow running.   we should delete PDB resource, once the workflow execution was done. This will optimize the etcd size and performance.

Reply:
I don't feel strongly about this. @jessesuen

Reply:
PDB does not look bit in size to me.

Reply:
Are you saying we cannot delete pods in there is a matching PDB?

Reply:
As discussed, we already delete resources on completion to keep things tidy, reduce etcd usage etc.